### ISSUE TITLE
ebgp-local_as plugin does not recognize bgp.local_as attribute

### ISSUE THREAD

**Issue by jbemmel** [ipspace/netlab#361]
The following does not work:
```
vrfs:
  red:
  blue:

vlans:
  red:
    mode: route
    vrf: red
  blue:
    mode: route
    vrf: blue
  vrf-leak:
    mode: route

nodes:
  r1:
  r2:
    module: [ vlan,vrf,ospf,bgp ]
    bgp.as: 65000
  r3:

links:
- r1:
  r2:
  vlan.trunk: [ red, blue ]
- r2:
  r3:
  vlan.trunk: [ red, blue ]
- interfaces: # VRF route leaking between red and blue, using BGP peering
  - node: r2
    vrf: red
    ipv6: True
    ipv4: False
    bgp.local_as: 65001
    vlan.access: vrf-leak
  - node: r2
    vrf: blue
    ipv6: True
    ipv4: False
    bgp.local_as: 65002
    vlan.access: vrf-leak
```

The r2 link loops to itself, causing it to get excluded from the neighbor list (non-VLAN links do have logic to handle this case)

**Comment by ipspace** [comment]
This seems to be a duplicate of #325 (or at least very similar to it). Also, even if the neighbor list would be OK, you'd need some extra plugin for the local_as to be recognized resulting in the desired EBGP session.

**Comment by jbemmel** [comment]
Yes it’s related, just highlighting an additional special case; could be added as a test case

**Comment by ipspace** [comment]
The "neighbor list" part should be fixed.

**Comment by ipspace** [comment]
**bgp.local_as** has been implemented in **ba65edca597f5e4aa0980df88bf9a90e5f70c1a1**. The above test case works on Cisco IOS and Arista EOS as long as we have an IPv4 session between VRFs.
